spot_driver_cpp/src/spot_image_publisher.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver_cpp/spot_image_publisher.hpp>
+
+#include <rclcpp/node.hpp>
+#include <rmw/qos_profiles.h>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_logger_interface.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_parameter_interface.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_publisher_interface.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_tf_interface.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_wall_timer_interface.hpp>
+#include <spot_driver_cpp/interfaces/spot_interface.hpp>
+#include <spot_driver_cpp/spot_image_sources.hpp>
+#include <spot_driver_cpp/types.hpp>
+
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <stdexcept>
+
+namespace
+{
+constexpr auto kImageCallbackPeriod = std::chrono::duration<double>{1.0 / 15.0 };  // 15 Hz
+constexpr auto kDefaultDepthImageQuality = 100.0;
```

Comment:
As far as I understand the `quality` setting has no effect for depth images, since Spot does not allow setting them to use JPEG compression. I'd need to test with a robot to see if we can omit this entirely, though.